heron/executor/src/python/heron_executor.py

```diff
@@ -174,17 +157,48 @@ def __init__(self, args):
     self.scheduler_classpath = args[30]
     self.scheduler_port = args[31]
 
-    # Read the heron_internals.yaml for cluster internal config
-    self.heron_internals_config = {}
-    with open(self.heron_internals_config_file, 'r') as stream:
-      self.heron_internals_config = yaml.load(stream)
-    self.log_dir = self.heron_internals_config['heron.logging.directory']
+    # Read the heron_internals.yaml for logging dir
+    self.log_dir = self.load_logging_dir(self.heron_internals_config_file)
+
+    # these get set when we call update_instance_distribution
+    self.instance_distribution = {}
+    self.stmgr_ids = []
+    self.metricsmgr_ids = []
+    self.heron_shell_ids = []
+
+    # this will soon be refactored to get this from the state manager instead of args
```

Comment:
Not directly related to this pull request. But I think it is better "this will soon be refactored to get this from the TMaster instead of args"